app/models/ems_refresh/save_inventory.rb

```diff
@@ -181,6 +181,29 @@ def save_vms_inventory(ems, hashes, target = nil)
     end
   end
 
+  # Convert all mapped hashes into actual tags and associate them with the object.
+  # The hash[:tags] object should be an array of hashes, probably created by
+  # the ContainerLabelTagMapping.map_labels method. Each hash in the array should
+  # have the following basic structure:
+  #
+  # {:category_tag_id=>139, :entry_name=>"foo", :entry_description=>"bar"}
+  #
+  # For now, the object must be a VmOrTemplate object.
+  #
+  def save_tags_inventory(object, hash)
+    return if hash.blank?
+    return unless object.kind_of?(VmOrTemplate)
+
+    tags = hash[:tags].collect do |tag_hash|
+      ContainerLabelTagMapping.find_or_create_tag(tag_hash)
```

Comment:
I see, it's in https://github.com/Ladas/manageiq/blob/b5f2f37f3beab86f3f71f127bb3c2ae0b0ea4294/app/models/ems_refresh/save_inventory_container.rb#L430-L430

Maybe we should just move this whole save_tags_inventory here?

@kbrock are you fine with using the ineffective way and then fixing the whole ContainerLabelTagMapping separately? Also we should extract it from ContainerLabelTagMapping, since it's a generic tagging problem, not just for Containers?